src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/Descriptors.cs

```diff
@@ -120,8 +120,8 @@ internal class Descriptors
 
         public static DiagnosticDescriptor AZC0018 = new DiagnosticDescriptor(
             nameof(AZC0018),
-            "Do ensure protocol method take a RequestContext parameter called context and not take models as parameter type or return type.",
-            "Protocol method should have requestContext as the last parameter and don't have model as parameter type or return type.",
+            "Do ensure protocol methods take a RequestContext parameter called `context` and do not take models as parameter or return types. Do ensure protocol methods will not cause an ambiguous overload resolution with convenience methods.",

```

Comment:
Would it make sense to have different errors for (1) requiring `context` and prohibiting model types and (2) identifying ambiguous overloads with convenience methods?  As a developer, I might want to handle these separately.

Reply:
I added a new rule AZC0019 for identifying ambiguity.